### PR TITLE
oracles-and-oracle-networks

### PR DESCRIPTION
**Pull request by Phenzic**
### Problem
- There are Grammar and Spelling errors in the piece
- Found broken links in the article
- The code snippet for the aggregator contract was updated in the source code
- Typescript unit test needed optimization, to reduce redundancy, handle error properly, 
  - it had inconsistent means of logging data
- The typescript contract call script (client-side) didn't have `anchor` imported
  

### Summary of Changes
I Made Grammar and code modifications to the article.

Fixes #
- I modified the aggregator contract based on the source code reference on github
- Resolved grammatic errors, 
- Made changes to foster concise reading
- Corrected Spell Checks
- Updated & replaced broken Links
- Modified typescript unit test to do the following: 
  - Added utility functions (getSwitchboardAggregator and deriveEscrowStateAccount) to reduce code duplication and improve readability.
  - I used assert.strictEqual and assert.isTrue for more precise assertions.
  - I Improved error handling with more specific assertions and clearer error messages.
  - I added consistent usage of console.error for logging errors and console.log for regular output.
  - I Improved overall the code readability by reducing redundant code and using clearer function and variable names.
  - I also strengthened the assertions to ensure that the expected outcomes are met in each test.
  - I also updated the headers to follow the SEO requirements (arranged in steps)
  
- For the Typescript contract call script:
  - I used the anchor.web3.Keypair.generate() method since its more conventional and readable when generating a new keypair in Solana.
  - I created a reusable function to use across your application. This will reduce overhead and potential issues with multiple network connections.
  - I added a console.log statement to display the fetched SOL price. 
  - I added a try-catch error handler for the fetchPrice function
  
Note: apologies for the 2 extra edited file; a little setback from using Prettify

### CODE REVIEW

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ objectives:
   - Explain why onchain programs cannot readily access real-world data on their
     own
   - Explain how oracles solve the problem of accessing real-world data onchain
-  - Explain how incentivized oracle networks make data more trustworthy
+  - Explain how incentivized Oracle networks make data more trustworthy
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:7]: oracle is lowercase.

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 2]
```diff
@@ -21,13 +21,13 @@ description: Access real-world data inside a Solana program.
...
 
 [Oracles](https://solana.com/ecosystem/explore?categories=oracle) are services
-that provide external data to a blockchain network. Blockchains by nature are
-siloed environments that do not know the outside world. This constraint
+that provide external data to a blockchain network. The blockchains, by nature,
+is a siloed network that do not interact with external data. This constraint
 inherently puts a limit on the use cases for decentralized applications (dApps).
 Oracles provide a solution to this limitation by creating a decentralized way to
-get real-world data onchain.
+get real-world off-chain data onchain.
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:28]: 'the blockchains is a siloed network' doesn't make sense.

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 3]
```diff
@@ -257,51 +263,94 @@ The actual onchain data for a Switchboard feed account looks a little like this:
 // https://github.com/switchboard-xyz/sbv2-solana/blob/0b5e0911a1851f9ca37042e6ff88db4cd840067b/rust/switchboard-solana/src/oracle_program/accounts/aggregator.rs#L60
 
 pub struct AggregatorAccountData {
-    /// Name of the aggregator to store onchain.
+    /// Name of the aggregator to store on-chain.
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:266]: No, see CONTRIBUTING.md

File: content/courses/connecting-to-offchain-data/oracles.md (hunk 1 above)
Comment (Phenzic) [reply content/courses/connecting-to-offchain-data/oracles.md:7]: I've made the change throughout the article

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 4]
```diff
@@ -414,28 +464,50 @@ You can also view the current value stored in an `AggregatorAccountData` account
...
+    }
+
+    console.log(`Latest SOL Price: ${solPrice.toString()}`);
+  } catch (error) {
+    console.error("Failed to fetch SOL price:", error);
+  }
 }
+
+// Call the function to fetch the SOL price
+fetchSolPrice();
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:510]: No need for wrapper functions. See CONTRIBUTING.md.

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 5]
```diff
@@ -414,28 +464,50 @@ You can also view the current value stored in an `AggregatorAccountData` account
 client-side in Typescript.
 
 ```typescript
-import { AggregatorAccount, SwitchboardProgram} from '@switchboard-xyz/solana.js'
+import {
+  AggregatorAccount,
+  SwitchboardProgram,
+} from "@switchboard-xyz/solana.js";
+import * as anchor from "@project-serum/anchor";
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:471]: `@project-serum/anchor` does not exist anymore.

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 6]
```diff
@@ -698,34 +775,35 @@ switchboard-v2 = "0.4.0"
 
 Before we get started with the logic, let’s go over the structure of our
 program. With small programs, it’s very easy to add all of the smart contract
-code to a single `lib.rs` file and call it a day. To keep it more organized
-though, it’s helpful to break it up across different files. Our program will
-have the following files within the `programs/src` directory:
+code to a single `lib.rs` file and call it a day. To keep it more organized,
+though, it’s helpful to break it up across different files.
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:779]: Use the `multiple` files template. This is mentioned in CONTRIBUTING.md.

File: content/courses/connecting-to-offchain-data/oracles.md [hunk 7]
```diff
@@ -1233,74 +1320,56 @@ describe("burry-escrow", () => {
...
+
       console.log("Onchain unlock price:", newAccount.unlockPrice);
       console.log("Amount in escrow:", escrowBalance);
 
       // Check whether the data onchain is equal to local 'data'
-      assert(failUnlockPrice == newAccount.unlockPrice);
-      assert(escrowBalance > 0);
+      assert.strictEqual(failUnlockPrice, newAccount.unlockPrice);
+      assert.isAbove(escrowBalance, 0);
     } catch (e) {
```
Comment (mikemaccana) [inline content/courses/connecting-to-offchain-data/oracles.md:1330]: Use full names. This is mentioned in `CONTRIBUTING.md`

File: tsconfig.json [hunk 8]
```diff
@@ -18,20 +18,20 @@
...
   "include": [
     ".contentlayer/generated",
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
-    ".next/types/**/*.ts"
+    ".next/types/**/*.ts",
   ],
-  "exclude": ["node_modules"]
+  "exclude": ["node_modules"],
```
Comment (mikemaccana) [inline tsconfig.json:36]: Remove these from the PR.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
This PR seems mainly to be minor grammar fixes, updates from Switchboard's GitHub gists, and doesn't meet the CONTRIBUTING.md requirements.

**Pull request closed without merging**